hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/replication/TestContainerImporter.java

```diff
@@ -0,0 +1,162 @@
+    long containerId = 1;
+    // create container
+    KeyValueContainerData containerData = new KeyValueContainerData(containerId,
+        ContainerLayoutVersion.FILE_PER_BLOCK, 100, "test", "test");
+    KeyValueContainer container = new KeyValueContainer(containerData, conf);
+    ContainerController controllerMock = mock(ContainerController.class);
+    // create containerImporter object
+    ContainerSet containerSet = new ContainerSet(0);
+    containerSet.addContainer(container);
+    MutableVolumeSet volumeSet = new MutableVolumeSet("test", conf, null,
+        StorageVolume.VolumeType.DATA_VOLUME, null);
+    ContainerImporter containerImporter = new ContainerImporter(conf,
+        containerSet, controllerMock, volumeSet);
+    File tarFile = new File("dummy.tar");
+    // second import should fail immediately
+    try {
+      containerImporter.importContainer(containerId, tarFile.toPath(),
+          null, NO_COMPRESSION);
+      assertFalse(true, "exception should occur");
+    } catch (StorageContainerException ex) {
+      assertTrue(ex.getResult().equals(
+          ContainerProtos.Result.CONTAINER_EXISTS));
+    }
+  }
+
+  @Test
+  void importSameContainerWhenFirstInProgress() throws Exception {
+    long containerId = 1;
+    // create container
+    KeyValueContainerData containerData = new KeyValueContainerData(containerId,
+        ContainerLayoutVersion.FILE_PER_BLOCK, 100, "test", "test");
+    KeyValueContainer container = new KeyValueContainer(containerData, conf);
+    // mock controller for return container data with delay
+    ContainerController controllerMock = mock(ContainerController.class);
+    Semaphore semaphore = new Semaphore(1);
+    when(controllerMock.importContainer(any(), any(), any()))
+        .thenAnswer((invocation) -> {
+          semaphore.acquire();
+          return container;
+        });
+    // create containerImporter object
+    ContainerSet containerSet = new ContainerSet(0);
+    MutableVolumeSet volumeSet = new MutableVolumeSet("test", conf, null,
+        StorageVolume.VolumeType.DATA_VOLUME, null);
+    ContainerImporter containerImporter = new ContainerImporter(conf,
+        containerSet, controllerMock, volumeSet);
+    // run import async first time having delay
+    File tarFile = containerTarFile(containerId, containerData);
+    CompletableFuture.runAsync(() -> {
+      try {
+        containerImporter.importContainer(containerId, tarFile.toPath(),
+            null, NO_COMPRESSION);
+      } catch (Exception ex) {
+        // do nothing
+      }
+    });
+    GenericTestUtils.waitFor(() -> semaphore.availablePermits() == 0,
+        10, 5000);
+    // run import second time and should fail immediately as
+    // first import in progress
+    try {
+      containerImporter.importContainer(containerId, tarFile.toPath(),
+          null, NO_COMPRESSION);
+      assertFalse(true, "exception should occur");
+    } catch (StorageContainerException ex) {
```

Comment:
To ensure test case is verifying the intended scenario.

```suggestion
    } catch (StorageContainerException ex) {
      assertTrue(ex.getMessage().contains("import in progress"),
          ex.getMessage());
```